### PR TITLE
Refactor/router consistency maintainability

### PR DESCRIPTION
**Pull request by bramirez96**
# Description

Making changes to improve the maintainability of our codebase. Any operations that are handled in consistent ways across different endpoints are being abstracted out into what is effectively middleware. This should make our code simpler to read as well as easier to maintain.

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

## Change Status

- [x] Complete, tested, ready to review and merge
- [ ] Complete, but not tested (may need new tests)
- [ ] Incomplete/work-in-progress, PR is for discussion/feedback

## Has This Been Tested

- [x] Yes
- [ ] No
- [ ] Not necessary

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] My code has been reviewed by at least one peer
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] There are no merge conflicts

### CODE REVIEW

**Review by trevorjamesmartin — APPROVED** [review]
well done!

File: api/avatar/avatarRouter.js [hunk 1]
```diff
@@ -1,8 +1,9 @@
 const router = require('express').Router();
-const authRequired = require('../middleware/authRequired');
+
+const { authRequired, avatarValidation, fileUpload } = require('../middleware');
+const { ops } = require('../../lib');
```
Comment (Nick-Ohman) [inline api/avatar/avatarRouter.js:4]: what is this?

Comment (bramirez96) [reply api/avatar/avatarRouter.js:4]: I changed the middleware to be a module-based export from the index file! And the ops module has functions for basic CRUD operations on data

Comment (Schrese) [reply api/avatar/avatarRouter.js:4]: This is such a cool idea

Comment (Nick-Ohman) [reply api/avatar/avatarRouter.js:4]: wow

**Review by Nick-Ohman — APPROVED** [review]
Looks great!

**Review by SandraCoburn — APPROVED** [review]
Looks great!

**Review by JLC6290 — APPROVED** [review]

File: api/stories/storyRouter.js [hunk 2]
```diff
@@ -113,18 +116,11 @@ router.get('/', authRequired, async (req, res) => {
...
-    if (story.length > 0) {
-      res.status(200).json(story[0]);
-    } else {
-      res.status(404).json({ error: 'StoryNotFound' });
-    }
-  } catch ({ message }) {
-    res.status(500).json({ message });
-  }
+
+  ops.getById(res, Stories.getById, 'Story', id);
```
Comment (Schrese) [inline api/stories/storyRouter.js:123]: This is super cool!

**Review by Schrese — APPROVED** [review]
looks good

**Pull request merged**